File: JavaScript solutions/347. Top K Frequent Elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
    let pp = [];
    if (nums.length === 0) return [];

    nums.sort((a, b) => a - b); 
    let cnt = 1;
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] === nums[i - 1]) {
            cnt++;
        } else {
            pp.push([cnt, nums[i - 1]]);
            cnt = 1;
        }
    }
    
    pp.push([cnt, nums[nums.length - 1]]);

    pp.sort((a, b) => b[0] - a[0]); 

    let ans = [];
    for (let i = 0; i < k && i < pp.length; i++) {
        ans.push(pp[i][1]);
    }
    return ans;
};

File: C++ solutions/347. Top K Frequent Elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<pair<int,int>> pp;
        if (nums.empty()) return {};

        sort(nums.begin(), nums.end()); 
        int cnt = 1;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] == nums[i - 1]) {
                cnt++;
            } else {
                pp.push_back(make_pair(cnt, nums[i - 1]));
                cnt = 1;
            }
        }
        
        pp.push_back(make_pair(cnt, nums[nums.size() - 1]));

        sort(pp.begin(), pp.end(), greater<pair<int,int>>()); 

        vector<int> ans;
        for (int i = 0; i < k && i < pp.size(); i++) {
            ans.push_back(pp[i].second);
        }
        return ans;
    }
};